test-data/unit/check-based-tuple-literal.test

```diff
@@ -0,0 +1,20 @@
+[case testTupleLiteralType]
+a: (int, str) = 1, ""
+reveal_type(a)  # N: Revealed type is "(int, str)"
+b: (str,) = ("sus",)
+reveal_type(b)  # N: Revealed type is "(str,)"
+c: () = (1,)  # E: Incompatible types in assignment (expression has type "(int,)", variable has type "()")  [assignment]
+reveal_type(c)  # N: Revealed type is "()"
+
+def f1() -> (int, str):
+    return 1, ""
+reveal_type(f1) # N: Revealed type is "(int, str)"
+
+def f2() -> (str,):
+    return "sus",
+reveal_type(f2) # N: Revealed type is "def() -> (str, )"
+
+def f3() -> ():
+    return 1,  # E: Incompatible return value type (got "(int,)", expected "()")  [return-value]
+reveal_type(f3) # N: Revealed type is "def () -> ()"
+[builtins fixtures/tuple.pyi]
```

Comment:
Can you add a test for using these types in 'value' positions?

```py
from typing import TypeVar, TypeAlias

A = TypeVar("A", bound=(int, str))

B: TypeAlias = (int, str)
```
Unfortunately, I expect these to fail. We can think about fixing this in a follow up.